scripts/path_plans.py

```diff
@@ -260,80 +150,217 @@ def _rotationMatrix_orient_z_axis(self, local_point):
+
+        plane_xz1_offset = np.array([-0.5*x_len,     0.5*self._size[1] + self._clearance,  0])
+        plane_xz2_offset = np.array([-0.5*x_len, -1*(0.5*self._size[1] + self._clearance), 0])
+        plane_yz1_offset = np.array([    0.5*self._size[0] + self._clearance,  -0.5*y_len, 0])
+        plane_yz2_offset = np.array([-1*(0.5*self._size[0] + self._clearance), -0.5*y_len, 0])
+
+        # Find how far we can tilt each plane, scale that based on self._slope
+        h = 0.5*self._size[2]
+        theta = self._slope*np.arctan(h / self._clearance)
+
+        # each plane's rotation
+        rot_xz1 = self.tf.create_rotation_matrix([theta], 'x')
+        rot_xz2 = self.tf.create_rotation_matrix([-1*theta], 'x')
+        rot_yz1 = self.tf.create_rotation_matrix([0.5*np.pi, -1*theta], 'zy')
+        rot_yz2 = self.tf.create_rotation_matrix([0.5*np.pi, theta], 'zy')
+
+        # combine into a transformation matrix to place planes around global origin
+        tf_xz1 = self.tf.get_transformation(rot_xz1, plane_xz1_offset)
+        tf_xz2 = self.tf.get_transformation(rot_xz2, plane_xz2_offset)
+        tf_yz1 = self.tf.get_transformation(rot_yz1, plane_yz1_offset)
+        tf_yz2 = self.tf.get_transformation(rot_yz2, plane_yz2_offset)
+
+        no_rotation = np.identity(3)
+        tf_final_offset = self.tf.get_transformation(no_rotation, [0, 0, -1*self._offset])
+
+        # Create rotation transformations
+        path_as_transforms = []
+        planes = [plane_xz1, plane_yz1, plane_xz2, plane_yz2]
+        for i, plane in enumerate(planes):
+            for p in plane.get_points():
+                # planes are just List[x,y,z], to move them around we make them info a vector
+                # [x, y, z, 1]' and apply homogenous transforms
+                p.append(1)
+                point = np.asarray(p).reshape(4)
+
+                # place around origin
+                if i == 0:  # xz1
+                    point_around_origin = np.matmul(tf_xz1, point)
+                elif i == 1:  # yz1
+                    point_around_origin = np.matmul(tf_yz1, point)
+                elif i == 2:  # xz2
+                    point_around_origin = np.matmul(tf_xz2, point)
+                else:  # 'yz2'
+                    point_around_origin = np.matmul(tf_yz2, point)
+
+                # Now that the points are spaced around the origin we can get their orientation
+                # to describe them as a full homogeneous matrix
+                xyz = point_around_origin[:-1]
+                point_rot = self._get_z_axis_oriented_rotation(xyz)
+                tf_around_origin = self.tf.get_transformation(point_rot, xyz)
+
+                # place around part and apply offset
+                tf_around_part = np.matmul(self._locator_tf, tf_around_origin)
+
+                # post to move relative to each frame instead of the world frame
+                tf_with_offset = np.matmul(tf_around_part, tf_final_offset)
```

Comment:
![image](https://user-images.githubusercontent.com/44533530/112757277-15a73980-8fb7-11eb-96a9-d0a9789a1edd.png)

Noting here the effects of applying offset along the camera frame's z-axis. The curved like shape is unsurprising given the technique we are using. I think it is acceptable as is.

If desired to change back to the entire plane moving, would need to change so offset is based on the fixed frame... but then you need to account for rotation as well. So probably not worth the hassle.

Reply:
Agreed. I think the only issue here is that it might be unexpected for an user unfamiliar with the code so if we leave it as it (which it sounds like we are) I think we should make sure it is documented.

And yeah, the other way is doable and would require us to assign each point a temporary frame where Z is normal to the plane, then offset, then assign the correct rotation matrix. But I don't feel like writing the code to do all that right now.

Reply:
I'm comfortable with no change moving forward. We'll need to document well. I might suggest doing such documentation in the Repo Wiki.